cdap-ui/app/cdap/components/AbstractWidget/FormInputs/Select/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import Select from '@material-ui/core/Select';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import InputBase from '@material-ui/core/InputBase';
+
+const styles = (theme) => {
+  return {
+    input: {
+      borderRadius: 4,
+      position: 'relative' as 'relative',
+      border: '1px solid #ced4da',
+      padding: '10px 18px 10px 12px',
+      '&:focus': {
+        backgroundColor: 'transparent',
+        borderRadius: 4,
+      },
+    },
+  };
+};
+const CustomizedInput = withStyles(styles)(InputBase);
+
+interface ISelectOptions {
+  value: string | number; // We need to expand this when we have complex use cases
+  label: string;
+}
+interface ISelectProps {
+  value: string;
+  onChange: (value: string) => void;
+  options: ISelectOptions[];
+  init: string; // initial selection?
+}
+
+export default function CustomSelect({ value, onChange, options }: ISelectProps) {
+  const onChangeHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const v = event.target.value;
+    if (v && typeof onChange === 'function') {
```

Comment:
this won't work if the select dropdown value is an empty string; which some of the dropdown in pipeline options actually have

Reply:
Have removed the condition.